npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -708,16 +689,63 @@ def raise_resolvability_error(error_message, yarn_lock)
         ).returns(T::Boolean)
       end
       def pattern_in_message(patterns, message)
-        patterns.any? do |pattern|
+        patterns.each do |pattern|
           if pattern.is_a?(String)
-            return message.include?(pattern)
+            return true if message.include?(pattern)
           elsif pattern.is_a?(Regexp)
-            message = message.gsub(/\e\[[\d;]*[A-Za-z]/, "")
-            return message.match?(pattern)
+            return true if message.gsub(/\e\[[\d;]*[A-Za-z]/, "").match?(pattern)
           end
         end
         false
       end
+
+      sig do
+        params(error_message: String, yarn_lock: Dependabot::DependencyFile)
+          .returns(T::Hash[T.any(Symbol, String), T.any(String, NilClass)])
+      end
+      def handle_package_not_found(error_message, yarn_lock) # rubocop:disable Metrics/PerceivedComplexity
+        # There are 2 different package not found error messages
+        p1 = error_message.include?(PACKAGE_NOT_FOUND)
+        p2 = error_message.match?(PACKAGE_NOT_FOUND2)
+
+        # If non of the patterns are found, return an empty hash
+        return {} unless p1 || p2
```

Comment:
can we use more descriptive names instead of p1 and p2?

Reply:
replaced to be more descriptive